src/gui/PeakControllerDialog.cpp

```diff
@@ -24,34 +24,25 @@
  */
 
 
-#include <QLabel>
-#include <QPushButton>
-#include <QMdiArea>
-#include <QPainter>
-
-#include "CaptionMenu.h"
-#include "gui_templates.h"
-#include "embed.h"
-#include "MainWindow.h"
-#include "ToolTip.h"
 
 #include "PeakController.h"
-#include "Knob.h"
-#include "TempoSyncKnob.h"
+
+#include <QPainter>
+
+#include "EffectControlDialog.h"
+#include "plugins/peak_controller_effect/peak_controller_effect.h"
+#include "plugins/peak_controller_effect/peak_controller_effect_control_dialog.cpp"
```

Comment:
Can't you include `"embed.h"` and `<QIcon>` and then change line 42 to:
```cpp
	setWindowIcon( QIcon(embed::getIconPixmap( "controller" )) );
```

instead of including `peak_controller_effect_control_dialog.cpp`?

Reply:
I don't understand. Please, can you elaborate? Have you tested your suggestion?

Reply:
Yes when I remove the .cpp include it fails compiling with: `error: 'embed' has not been declared`. 

So I added `#include "embed.h"`. Then it fails because  `setWindowIcon` aspects a `QIcon` but a `QPixmap` is given. 

So that's why also included `<QIcon>` and changed the `setWindowIcon` line.

After that it compiles and works fine here.